mm2src/coins/utxo/slp.rs

```diff
@@ -26,14 +28,90 @@ struct SlpToken {
     platform_utxo: UtxoStandardCoin,
 }
 
-#[derive(Deserializable, Default)]
+/// https://slp.dev/specs/slp-token-type-1/#transaction-detail
+#[derive(Debug, Eq, PartialEq)]
+enum SlpTransaction {
+    /// https://slp.dev/specs/slp-token-type-1/#send-spend-transaction
+    Genesis {
+        token_ticker: String,
+        token_name: String,
+        token_document_url: String,
+        token_document_hash: Vec<u8>,
+        decimals: Vec<u8>,
+        mint_baton_vout: Vec<u8>,
+        initial_token_mint_quantity: Vec<u8>,
+    },
+    /// https://slp.dev/specs/slp-token-type-1/#send-spend-transaction
+    Send { token_id: Vec<u8>, amount: Vec<u8> },
+}
+
+impl Deserializable for SlpTransaction {
+    fn deserialize<T>(reader: &mut Reader<T>) -> Result<Self, Error>
+    where
+        Self: Sized,
+        T: std::io::Read,
+    {
+        let transaction_type: String = reader.read()?;
+        match transaction_type.as_str() {
+            "GENESIS" => {
+                let token_ticker = reader.read()?;
+                let token_name = reader.read()?;
+                let maybe_push_op_code: u8 = reader.read()?;
+                let token_document_url = if maybe_push_op_code == Opcode::OP_PUSHDATA1 as u8 {
+                    reader.read()?
+                } else {
+                    let mut url = vec![0; maybe_push_op_code as usize];
+                    reader.read_slice(&mut url)?;
+                    String::from_utf8(url).map_err(|e| Error::Custom(e.to_string()))?
+                };
+
+                let maybe_push_op_code: u8 = reader.read()?;
+                let token_document_hash = if maybe_push_op_code == Opcode::OP_PUSHDATA1 as u8 {
+                    reader.read_list()?
+                } else {
+                    let mut hash = vec![0; maybe_push_op_code as usize];
```

Comment:
Just a question: why the `token_document_hash` can be written as [OP_PUSHDATA1, length, bytes...] and [length, bytes] both? Is it possible for `maybe_push_op_code` to be `OP_PUSHDATA2` or `OP_PUSHDATA4`?

Reply:
As per SLP docs https://slp.dev/specs/slp-token-type-1/#formatting, 
`
5. Some fields permit an empty push (length-0 chunk) however you cannot use opcode 0x00 for this purpose. Rather, you may use 0x4c 0x00, or 0x4d 0x00 0x00, or 0x4e 0x00 0x00 0x00 0x00.
`

Reply:
I will support additional PUSH opcodes too later.